### ISSUE TITLE
[Bug Report] invalid acosh backward result

### ISSUE THREAD

**Issue by hschoi4448** [tenstorrent/tt-metal#6583]
**Describe the bug**

The `acosh_bw` function returns an invalid gradient value.

**To Reproduce**
1. Copy and past below code to /tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_acosh.py 
```Python
# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.

# SPDX-License-Identifier: Apache-2.0

import torch
import pytest
import tt_lib
from tests.tt_eager.python_api_testing.unit_testing.backward_ops.utility_funcs import data_gen_pt_tt, compare_results

def data_gen_pt_tt(input_shapes, device, required_grad=False, val=1):
    pt_tensor = (torch.ones(input_shapes, requires_grad=required_grad) * val).bfloat16()
    tt_tensor = (
        tt_lib.tensor.Tensor(pt_tensor, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
    )
    return pt_tensor, tt_tensor

@pytest.mark.parametrize(
    "input_shapes",
    (
        (torch.Size([1, 1, 32, 32])),
    ),
)
def test_bw_acosh(input_shapes, device):
    in_data, input_tensor = data_gen_pt_tt(input_shapes, device, True, val=0.5)
    grad_data, grad_tensor = data_gen_pt_tt(input_shapes, device, False, val=1)

    print("input_tensor", input_tensor)
    print("grad_tensor", grad_tensor)
    
    pyt_y = torch.acosh(in_data)

    tt_output_tensor_on_device = tt_lib.tensor.acosh_bw(grad_tensor, input_tensor)

    in_data.retain_grad()

    pyt_y.backward(gradient=grad_data)

    golden_tensor = [in_data.grad]

    comp_pass = compare_results(tt_output_tensor_on_device, golden_tensor)
    
    print("tt_output_tensor_on_device", tt_output_tensor_on_device)
    print("golden_tensor", golden_tensor)
    assert comp_pass

```
2. Run cmd `pytest ./tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_acosh.py
```Python
input_tensor ttnn.Tensor([[[[ 0.50000,  0.50000,  ...,  0.50000,  0.50000],
               [ 0.50000,  0.50000,  ...,  0.50000,  0.50000],
               ...,
               [ 0.50000,  0.50000,  ...,  0.50000,  0.50000],
               [ 0.50000,  0.50000,  ...,  0.50000,  0.50000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
grad_tensor ttnn.Tensor([[[[ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               ...,
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
tt_output_tensor_on_device [ttnn.Tensor([[[[inf     , inf     ,  ..., inf     , inf     ],
               [inf     , inf     ,  ..., inf     , inf     ],
               ...,
               [inf     , inf     ,  ..., inf     , inf     ],
               [inf     , inf     ,  ..., inf     , inf     ]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)]
golden_tensor [tensor([[[[nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          ...,
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan]]]], dtype=torch.bfloat16)]
```

**Expected behavior**

I want `acosh_bw` to return the correct gradient

**Screenshots**

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Version of software (eg. commit 2c776dbd4f4fe9e26fa1e65216db3690a8a800c8)

**Additional context**

**Comment by umadevimcw** [comment]
@hschoi4448 Without any change in code I am getting following output

<img width="1181" alt="image" src="https://github.com/tenstorrent-metal/tt-metal/assets/29221091/87a84301-a652-41df-9510-a87312602671">

**Comment by hschoi4448** [comment]
I'm using `wormhole_b0`. I will also test it in grayskull.

![image](https://github.com/tenstorrent-metal/tt-metal/assets/48243030/6436e859-3c38-4a17-9fd4-1d3ae6ce2a03)

**Comment by umadevimcw** [comment]
@tt-aho @eyonland @jliangTT In WHB0 we are unable to store nan in tensor. The above is with respect to storing the nan. For GS the some code works fine. Need input on this. 

How to address this issue? or Can I close this issue considering the hardware limitations?

**Comment by umadevimcw** [comment]
@hschoi4448 @razorback3 https://github.com/tenstorrent/tt-metal/issues/8944, https://github.com/tenstorrent/tt-metal/issues/8945#issuecomment-2146247945, Please look at this comments for this issue

**Comment by VirdhatchaniKN** [comment]
Hi @ttmtrajkovic @tt-aho @razorback3 
Based on this comment here : [Link](https://github.com/tenstorrent/tt-metal/pull/11243/files#r1808777487)
I've fixed acosh_bw with the tech report values and would like your review on this PR - https://github.com/tenstorrent/tt-metal/pull/13983 (I will be removing the changes in the old merged PR in a separate PR if this is good to go)

Would like your reviews here to proceed with issue https://github.com/tenstorrent/tt-metal/issues/7488

**Comment by VirdhatchaniKN** [comment]
Fix : https://github.com/tenstorrent/tt-metal/pull/13983

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/pull/19695
PR Fixes issue, Will close after merge
